src/printer.js

```diff
@@ -1,4 +1,30 @@
 // https://prettier.io/docs/en/plugins.html#printers
-const genericPrint = (path, options) => options.originalText;
+const { concat, hardline, join } = require('prettier').doc.builders;
+
+const space = ' ';
+
+const genericPrint = (path, options, print) => {
+  const node = path.getValue();
+
+  switch (node.type) {
+    case 'SourceUnit':
+      return join(hardline, path.map(print, 'children'));
+    case 'PragmaDirective':
+      return concat([
+        'pragma',
+        space,
+        node.name,
+        space,
+        node.value,
+        ';',
+        hardline
+      ]);
+    default:
+      return concat([
```

Comment:
at the moment this default returns the source code as it is, I've had to add a `+ 1` to the end index and a `hardline` at the very end

Reply:
Awesome! This will start being used less and less as you finish the parser. 😎